src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs

```diff
@@ -189,6 +201,48 @@ public static TypedData ToRpc(this object value, ILogger logger, Capabilities ca
                     }
                 }
             }
+            else if (IsTypeDataCollectionSupported(capabilities) && value is byte[][] arrBytes)
+            {
```

Comment:
Change to 
```
else if (IsTypeDataCollectionSupported(capabilities))
            {
                typedData = ToRpcCollection(value);
            }
```
And add
```
internal static TypedData ToRpcCollection(this object value)
        {
            TypedData typedData = new TypedData();
            if (value is byte[][] arrBytes)
            {
                TypedDataCollectionBytes collectionBytes = new TypedDataCollectionBytes();
                foreach (byte[] element in arrBytes)
                {
                    if (element != null)
                    {
                        collectionBytes.Bytes.Add(ByteString.CopyFrom(element));
                    }
                }
                typedData.CollectionBytes = collectionBytes;
            }
            else if (value is string[] arrString)
            {
                TypedDataCollectionString collectionString = new TypedDataCollectionString();
                foreach (string element in arrString)
                {
                    if (!string.IsNullOrEmpty(element))
                    {
                        collectionString.String.Add(element);
                    }
                }
                typedData.CollectionString = collectionString;
            }
            else if (value is double[] arrDouble)
            {
                TypedDataCollectionDouble collectionDouble = new TypedDataCollectionDouble();
                foreach (double element in arrDouble)
                {
                    collectionDouble.Double.Add(element);
                }
                typedData.CollectionDouble = collectionDouble;
            }
            else if (value is long[] arrLong)
            {
                TypedDataCollectionSInt64 collectionLong = new TypedDataCollectionSInt64();
                foreach (long element in arrLong)
                {
                    collectionLong.Sint64.Add(element);
                }
                typedData.CollectionSint64 = collectionLong;
            }
            return typedData;
        }
```

Reply:
I thought about this way but this will not achieve what we are looking for. In case we have IsTypeDataCollectionSupported and the type is not any of the collection types then we lost default  JSON else type in the bigger nested if. I will think a way that is satisfy your request.